src/targets/public/index.jsx

```diff
@@ -0,0 +1,130 @@
+import { getDataset, getDataOrDefault } from 'lib/initFromDom'
+
+import flag from 'cozy-flags'
+
+const manifest = require('../../../manifest.webapp')
+
+const frenchAtlaskitCozy = require(`locales/atlassian_missing_french.json`)
+
+const locales = {
+  en: {
+    react: require('react-intl/locale-data/en'),
+    atlaskit: en
+  },
+  fr: {
+    react: require('react-intl/locale-data/fr'),
+    atlaskit: { ...fr, ...frenchAtlaskitCozy }
+  }
+}
+
+/*
+With MUI V4, it is possible to generate deterministic class names.
+In the case of multiple react roots, it is necessary to disable this
+feature. Since we have the cozy-bar root, we need to disable the
+feature.
+https://material-ui.com/styles/api/#stylesprovider
+*/
+const generateClassName = createGenerateClassName({
+  disableGlobal: true
+})
+
+const renderApp = function(appLocale, client) {
+  const App = require('components/app').default
+
+  render(
+    <WebviewIntentProvider>
+      <I18n
+        lang={appLocale}
+        dictRequire={appLocale => require(`locales/${appLocale}`)}
+      >
+        <StylesProvider generateClassName={generateClassName}>
+          <IntlProvider
+            locale={appLocale}
+            messages={locales[appLocale].atlaskit}
+          >
+            <CozyProvider client={client}>
+              <MuiCozyTheme>
+                <IsPublicContext.Provider value={true}>
+                  <App isPublic={true} />
+                </IsPublicContext.Provider>
+              </MuiCozyTheme>
+            </CozyProvider>
+          </IntlProvider>
+        </StylesProvider>
+      </I18n>
+    </WebviewIntentProvider>,
+    document.querySelector('[role=application]')
+  )
+}
+
+// initial rendering of the application
+export const initApp = () => {
+  const data = getDataset()
+
+  const appSlug = getDataOrDefault(data.app.slug, manifest.slug)
+  const appVersion = getDataOrDefault(data.app.version, manifest.version)
+
+  const supportedLocales = ['en', 'fr']
+
+  addLocaleData(locales.en.react)
+  addLocaleData(locales.fr.react)
+
+  const userLocale = getDataOrDefault(data.locale, 'en')
+  const appLocale = supportedLocales.includes(userLocale) ? userLocale : 'en'
+
+  const protocol = window.location ? window.location.protocol : 'https:'
+
+  const token = data.token
+
+  // initialize the client to interact with the cozy stack
+  const client = new CozyClient({
+    uri: `${protocol}//${data.domain}`,
+    token: token,
+    appMetadata: {
+      slug: appSlug,
+      version: appVersion
+    }
+  })
+  if (!Document.cozyClient) {
+    Document.registerClient(client)
+  }
+  client.registerPlugin(RealtimePlugin)
+  client.registerPlugin(flag.plugin)
+
+  return { appLocale, client }
+}
+
+const memoizedInit = memoize(initApp)
+const init = () => {
+  const { appLocale, client } = memoizedInit()
+  renderApp(appLocale, client)
+}
+document.addEventListener('DOMContentLoaded', init)
+
+if (module.hot) {
+  memoizedInit()
+  module.hot.accept()
+}
```

Comment:
Why do you have copy / pasted this thing, can't we use the other index.jsx?

Reply:
@Crash-- to get an entrypoint so webpack generate an bundle for public which wasn't the case before. I agree with you that we can remove `module.hot` but I will check if removing hot reload doesn't have any impact in watch mode on public page development.